### PR TITLE
fix(core/bitcoin): display descriptors for taproot XPUBs

### LINKED ISSUE #3475: Display descriptors for BTC Taproot accounts
Discussed [here](https://satoshilabs.slack.com/archives/CL1D61PQF/p1702899199080769).

### PR DESCRIPTION
**Pull request by mmilata**
Fixes #3475. I'm assuming there's no need to add new flag to GetPublicKey since non-descriptor Taproot XPUBs are non-standard? I'm also assuming we're not displaying the descriptor checksum since it's not on the suite screenshot in #3475.

Unfortunately on T2B1 the changes [increase the number of screens to three:](https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/6057776893/artifacts/master_diff/diff/TR-device_tests-bitcoin-test_descriptors.py%3A%3Atest_descriptors%5BBitcoin-1-86-InputScriptType.SPENDTAPROOT-tr%28-3a85f3dd.html). ddc50e277f04cd82e96ddc92a3dfe9a3f5641bcc3f05ed3e7e3b7b0d96f0b7a6

CI note: master diff is broken because we're skipping T2B1 tests so the screens are missing. [GitLab master diff](https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/6057776893/artifacts/master_diff/index.html).

### CODE REVIEW

**Comment by prusnak** [comment]
> I'm also assuming we're not displaying the descriptor checksum

Can we append the checksum and also update the Suite to also show the checksum?

**Comment by Hannsek** [comment]
Yes, we can totally do that if wanted. cc @Hermez-cz

**Comment by mmilata** [comment]
Added checksum in d724613.
[master diff](https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/6067805944/artifacts/master_diff/index.html)
Number of screens remains the same but it makes the CoinJoin QR code a bit larger but still probably usable:
Screenshot 2024-02-01 at 00-53-38 TT-device_tests-bitcoin-test_descriptors py test_descriptors Bitcoin-0-10025-InputScriptType SPENDTAPROOT--301d7568

**Comment by Hannsek** [comment]
Main diff seems to be empty?

**Comment by mmilata** [comment]
Try this one for now: https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/6067805944/artifacts/master_diff/index.html

**Comment by mmilata** [comment]
Screen-by-screen: https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/6067805944/artifacts/master_diff/master_diff.html

**Comment by obrusvit** [comment]
As Martin pointed out in the first comment, the number of screens on TS3 changed to 3 and it's not matching the Suite.
cc: @Hannsek 
image

**Comment by Hannsek** [comment]
Suite has to add checksum behind the descriptor, they'll hopefully fix that.

File: core/src/apps/bitcoin/common.py [hunk 1]
```diff
@@ -212,3 +212,48 @@ def format_fee_rate(
         shortcut = ""
 
     return f"{fee_rate_formatted} sat{shortcut}/{'v' if coin.segwit else ''}B"
+
+
+def descriptor_checksum(desc: str) -> str:
```
Comment (obrusvit) [inline core/src/apps/bitcoin/common.py:217]: Can we make a comment here that the function is identical to the one found in `src/trezorlib/tools.py`, just for cross-reference.

File: core/src/apps/bitcoin/get_public_key.py [hunk 2]
```diff
@@ -110,3 +116,23 @@ async def get_public_key(
...
         root_fingerprint=keychain.root_fingerprint(),
     )
+
+
+def _xpub_descriptor(
+    node_xpub: str,
+    address_n: list[int],
+    script_type: InputScriptType,
+    fingerprint: int,
+):
```
Comment (obrusvit) [inline core/src/apps/bitcoin/get_public_key.py:126]: missing type hint `-> str` but I wonder why style check doesn't complain :thinking:

**Review by obrusvit — COMMENTED** [review]
Apart from minor issues, LGTM. But Suite screen must be synced.

File: core/src/apps/bitcoin/common.py (hunk 1 above)
Comment (mmilata) [reply core/src/apps/bitcoin/common.py:218]: Good idea, fixed in [5b8865a](https://github.com/trezor/trezor-firmware/pull/3505/commits/5b8865a4a94f5bfc21c01ed6b2b6feda5ff41ea0).

File: core/src/apps/bitcoin/get_public_key.py (hunk 2 above)
Comment (mmilata) [reply core/src/apps/bitcoin/get_public_key.py]: That is weird, good catch, fixed in [5b8865a](https://github.com/trezor/trezor-firmware/pull/3505/commits/5b8865a4a94f5bfc21c01ed6b2b6feda5ff41ea0).

**Review by obrusvit — APPROVED** [review]

**Comment by bosomt** [comment]
**QA OK**

it will be retested after Trezor Suite PR is done and merged

**Info:**
 - Suite version: desktop 24.2.2 (6544eb4cdb6f8982f3f391307e2bff5784db2bcf)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.2.2 Chrome/118.0.5993.129 Electron/27.0.4 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T2B1 2.6.5 regular (revision e1f696b4dd8c024a9ceca276d7336b4d93ae7ff2)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**